### DISCUSSION TITLE
Config-File for internal Mail Server Settings

### DISCUSSION THREAD

**Discussion by macomar45** [ciur/papermerge#477] [Q&A]
Hello,
we have installed PM via Docker on Debian 11 (vSphere 7 and SAN Storage).
Now we are looking for the config file where the own internal mailserver for a password recovery of a user is entered.
In the online documentation is nothing noted.
Thanks a lot.

![2022-09-08 12_46_36-Papermerge _ Log in und 3 weitere Seiten - Profil 1 – Microsoft​ Edge](https://user-images.githubusercontent.com/113089427/189103461-1eae005d-b31f-4c14-a067-608224e012bd.png)

**Comment by w4tzmann** [comment]
Good Question!
With 2.1.0b I can't see any option the reset the password on the login screen.
With 2.0.0 I get an 500 Error, after entering my e-mail address. Unfortunately it seems that there is not log entry for this error.
With Debug: True, I get this error:
Exception Type: | OSError
[Errno 99] Cannot assign requested address

I have added the following lines to  "/opt/app/config/settings/production.py" . I'm able to send password reset mails.
```
EMAIL_HOST = "" #Mailserver IP address
EMAIL_PORT = "587" #Port from your Mailserver
EMAIL_HOST_USER = "" #mail account username
EMAIL_HOST_PASSWORD = "" #mail account password
EMAIL_USE_TLS = True #Because no one likes unsecure connections
DEFAULT_FROM_EMAIL  = "" #Default Sender address, for me the same as username
```

The disadvantage is that you have to set this setting again once you remove the container.

@ciur  is there any simple option, to add something like this?
```
docker-compose.yml
services:
   app:
      environment:
         - "EMAIL_HOST=" #Mailserver IP address
         - "EMAIL_PORT=" #Port from your Mailserver
         - "EMAIL_HOST_USER=" #mail account username
         - "EMAIL_HOST_PASSWORD=" #mail account password
         - "EMAIL_USE_TLS=" #Because no one likes unsecure connections
         - "DEFAULT_FROM_EMAIL=" #Default Sender address, for me the same as username

productions.py
EMAIL_HOST = os.environ.get('EMAIL_HOST')
EMAIL_PORT = os.environ.get('EMAIL_PORT')
EMAIL_HOST_USER = os.environ.get('EMAIL_HOST_USER')
EMAIL_HOST_PASSWORD = os.environ.get('EMAIL_HOST_PASSWORD')
EMAIL_USE_TLS = os.environ.get('EMAIL_USE_TLS')
DEFAULT_FROM_EMAIL  = os.environ.get('DEFAULT_FROM_EMAIL')
```
